ros/src/tl_detector/tl_detector.py

```diff
@@ -100,8 +198,71 @@ def get_closest_waypoint(self, pose):
             int: index of the closest waypoint in self.waypoints
 
         """
-        #TODO implement
-        return 0
+        #TODO This function is currently O(n). It can be improved to O(log n)
+        if self.waypoints_stamped is None:
+            return None
+
+        dist_min = sys.maxsize
+        wp_min = None
+
+        for wp in range(len(self.waypoints_stamped.waypoints)):
+            dist = self.distance2(pose, self.waypoints_stamped.waypoints[wp].pose.pose)
+
+            if dist < dist_min:
+                dist_min = dist
+                wp_min = wp
+
+        return wp_min
+
+    def extract_image(self, pred_image_mask, image):
```

Comment:
@ooleksyuk , is this working well when there are three traffic lights in a image?